client/unittest/ReportIssueDataCollector.test.ts

```diff
@@ -0,0 +1,94 @@
+/* --------------------------------------------------------------------------------------------
+ * Copyright (c) Microsoft Corporation. All rights reserved.
+ * Licensed under the MIT License. See License.txt in the project root for license information.
+ * ------------------------------------------------------------------------------------------ */
+
+import * as assert from 'assert';
+import { ReportIssueDataCollector } from 'microsoft.aspnetcore.razor.vscode/dist/Diagnostics/ReportIssueDataCollector';
+import { RazorLogger } from 'microsoft.aspnetcore.razor.vscode/dist/RazorLogger';
+import { Trace } from 'microsoft.aspnetcore.razor.vscode/dist/Trace';
+import * as vscode from 'microsoft.aspnetcore.razor.vscode/dist/vscodeAdapter';
+import { TestEventEmitterFactory } from './Mocks/TestEventEmitterFactory';
+import { TestTextDocument } from './Mocks/TestTextDocument';
+import { createTestVSCodeApi } from './Mocks/TestVSCodeApi';
+
+describe('ReportIssueDataCollector', () => {
+
+    it('start always logs the starting of data collection', async () => {
+        // Arrange
+        const api = createTestVSCodeApi();
+        const razorOutputChannel = api.getRazorOutputChannel();
+        const eventEmitterFactory = new TestEventEmitterFactory();
+        const logger = new RazorLogger(api, eventEmitterFactory, Trace.Off);
+        const eventEmitter = eventEmitterFactory.create<vscode.TextDocument>();
+        const dataCollector = new ReportIssueDataCollector(eventEmitter.event, logger);
+
+        // Act
+        dataCollector.start();
+
+        // Assert
+        const lastLog = razorOutputChannel[razorOutputChannel.length - 1].trim();
```

Comment:
Super-minor nitpick: the `.trim()` isn't needed.

Also it's a little odd that the condition on the next line is `> 0` not `>= 0`. Are we deliberately specifying it must not be at the start of the string, or is the intention to check the substring is present anywhere?

I know this is not a big deal.

Reply:
> Super-minor nitpick: the `.trim()` isn't needed.

👍 

> Also it's a little odd that the condition on the next line is `> 0` not `>= 0`. Are we deliberately specifying it must not be at the start of the string, or is the intention to check the substring is present anywhere?

I'm assuming it's not at the start of the line because we also expect that logs have a date time stamp prefix. That being said, these tests shouldn't be concerned with that so I'll definitely change it for the sake of clarity.